### DISCUSSION TITLE
how choose Rocket?

### DISCUSSION THREAD

**Discussion by ChuanDou2021** [rwf2/Rocket#2285] [Questions]
why rocket providing 2 versions: 0.4 & 0.5, what's different? how choose? 
thanks

**Comment by robertwayne** [comment]
[You should skim this page.](https://rocket.rs/v0.5-rc/guide/upgrading/) The CHANGELOG also contains all the actual details. One of the major differences is 0.5 is powered by Tokio (async), while 0.4 is sync.

0.5 isn't actually released yet, it's a "release candidate" - meaning pre-release. Eventually 0.5 will become the primary Rocket version.

I would personally recommend starting with 0.5.0-rc.2 _(the latest release candidate)_. It has proven very stable for me _(I use it on production with no concerns)_, and will reduce the friction of upgrading in the future. Obviously evaluate your own use-case first.

**Reply by ChuanDou2021** [reply]
Thanks, I got it.
Great work, we love rocket
